File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.cpp
#include <string>
#include <algorithm>
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(string s) {
    int s_l = s.size();
    vector<int> answer;
    vector<int> position(24, 0); // -97
    for (int i = 0; i < s_l; i++){
        //cout << i << " " << s[i] << " " << (int)s[i] << " " << (int)s[i]-97 << endl;
        if (position[(int)s[i]-97] > 0) answer.push_back(i-position[(int)s[i]-97]+1);
        else answer.push_back(-1);
        position[(int)s[i]-97] = i+1;   
        //cout << position[(int)s[i]-97] << endl;
    }
    return answer;
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    position = [0]*26
    for idx, c in enumerate(s):
        if position[ord(c)-97] : answer.append(idx-position[ord(c)-97]+1)
        else: answer.append(-1)
        position[ord(c)-97] = idx+1
    return answer